### PR TITLE
Fix label_when_reviewed_workflow_run permissions

### PR DESCRIPTION
**Pull request by ashb**
This workflow run is currently failing with:

```
Run ./.github/actions/checks-action
  with:
    token: ***
    name: Selective build check
    status: in_progress
    sha: 2cf8c7f268c1db73d840f029aa5180941519c492
    details_url: https://github.com/apache/airflow/actions/runs/960898933
    output: {"summary": "Checking selective status of the build in [the run](https://github.com/apache/airflow/actions/runs/960898933) "}

Error: Resource not accessible by integration
```

so I _think_ this will help

---
**^ Add meaningful description above**

Read the **[Pull Request Guidelines](https://github.com/apache/airflow/blob/main/CONTRIBUTING.rst#pull-request-guidelines)** for more information.
In case of fundamental code change, Airflow Improvement Proposal ([AIP](https://cwiki.apache.org/confluence/display/AIRFLOW/Airflow+Improvements+Proposals)) is needed.
In case of a new dependency, check compliance with the [ASF 3rd Party License Policy](https://www.apache.org/legal/resolved.html#category-x).
In case of backwards incompatible changes please leave a note in [UPDATING.md](https://github.com/apache/airflow/blob/main/UPDATING.md).

### CODE REVIEW

**Comment by ashb** [comment]
@potiuk PTAL -- I think this is the missing permission, but the error doesn't really provide much/any clue.

Failing example https://github.com/apache/airflow/runs/2885563848?check_suite_focus=true

File: .github/workflows/label_when_reviewed_workflow_run.yml [hunk 1]
```diff
@@ -23,6 +23,7 @@ on:  # yamllint disable-line rule:truthy
     types: ['requested']
 permissions:
   # All other permissions are set to none
+  chekcs: write
```
Comment (kaxil) [inline .github/workflows/label_when_reviewed_workflow_run.yml]: is this a typo or it is how it is?

Comment (potiuk) [reply .github/workflows/label_when_reviewed_workflow_run.yml]: yeah. Typo I think

Comment (potiuk) [reply .github/workflows/label_when_reviewed_workflow_run.yml]:
```suggestion
  checks: write
```

Comment (ashb) [reply .github/workflows/label_when_reviewed_workflow_run.yml]: Whoops yes.

**Review by kaxil — APPROVED** [review]

**Pull request merged**